### DISCUSSION TITLE
Outline and cloudflare tunnels

### DISCUSSION THREAD

**Discussion by roleforgaming** [outline/outline#7689] [Self Hosting]
After spending nearly  24+ hours trying to get this installed, I am just drawing a  blank. I've scrounged up all the resources I can find, but have yet to find anyone who has documented installing outline via docker and using cloudflare tunnels to proxy it... The closest I got ended up with outline at least loading the login screen, but then I could not reach the keycloak auth endpoint due to some issue with cloudflare... I don't understand why this is so complicated to spin up. Any help would be greatly appreciated... my brain is officially fried.

**Comment by tommoor** [comment]
You haven't provided any technical details at all, so I'm not sure what help you expect?

Why are cloudflare tunnels involved here? Seems irrelevant

**Reply by aaronsdevera** [reply]
It's just a useful proxy to use for self-hosting for personal and small zero trust networks :)

**Comment by edbourque0** [comment]
I have the same behavior and feeling on my side, but I can provide more technical details. I have the instance spinning in docker and can access it from my local ip (10.0.0.X:43002). I set up my cloudflare tunnel to foward wiki.mydomain.com to 10.0.0.X:43002 but when I try to access it thru https://wiki.mydomain.com it only shows a black screen without the login buttons (i do see those button when I access it thru my local ip). When I'm looking at the network diagnostics from my browser I can't see anything abnormal and container logs shows that it sent the request back.

Any ideas for troubleshooting ? I have a lot of containers running with this exact setup and never had an issue like that. 

Thanks in advance!

**Comment by edbourque0** [comment]
Update : Got it to work @roleforgaming!

I only needed to disable Cloudflare's Rocket Loader for my domain.
You can do this by going to your Cloudflare dashboard under Speed -> Optimizations -> Content Optimization -> Rocket Loader

Let me know if that worked for you!

**Reply by JosTheDude** [reply]
Thank you so much <3

**Comment by aaronsdevera** [comment]
Also running into this, the optimization bit didn't seem to work.

**Reply by aaronsdevera** [reply]
for now I modified `FORCE_HTTPS` to `false` and am routing the cloudflare tunnel to port 3000 for the outline container, essentially removing https from the container -> cloudflare tunnel connection

**Reply by rzm25** [reply]
Do you have authentication set up? Do you have DNS records created and applied?
